### PR TITLE
Display hearing type on case details

### LINKED ISSUE #15021: at the top of the case details page, display the hearing type
Connects #12826

## Description
The appeal type is displayed in a section at the top of the case details page under the label **TYPE**:

Screen Shot 2020-08-20 at 12 23 32 AM

Add a new section to the right of the appeal type (or appeal stream type, if #12826 has been merged) that displays the hearing type on the appeal.

Screen Shot 2020-08-20 at 12 41 32 AM

If it's a legacy appeal, and no hearing is scheduled, display the `LegacyAppeal.sanitized_hearing_request_type`.

Otherwise, show the type of the most recently scheduled hearing.

## Acceptance criteria
- [ ] the hearing type displays for ama and legacy appeals in the hearing docket
- [ ] update (or create) a storybook component for the CaseTitleDetails component
- [ ] update (or create) a feature test for the correct values in the new section

## Technical notes
[CaseTitleDetails](https://github.com/department-of-veterans-affairs/caseflow/blob/master/client/app/queue/CaseTitleDetails.jsx) is the component that contains the appeal type label and value.

### PR DESCRIPTION
**Pull request by rubaiyat22**
Resolves #15021

### Description
- create a new method `readable_hearing_request_type` on `Appeal` and `LegacyAppeal` which returns `Central`, `Video`, `Travel` or `nil` and serialize the method
- display the type on case details page depending on the docket type
- create a basic story for `CaseTitleDetails` component
- write unit tests in `appeal_spec` and feature tests in `case_details_spec`

### Acceptance Criteria
- [x] the hearing type displays for ama and legacy appeals in the hearing docket
- [x] update (or create) a storybook component for the CaseTitleDetails component
- [x] update (or create) a feature test for the correct values in the new section

### User Facing Changes
 - [x] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---
Screen Shot 2020-09-03 at 7 37 25 PM| Screen Shot 2020-09-03 at 7 36 37 PM

### Storybook Story
*For Frontend (Presentationa) Components*
* [x] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [x] Give it a title that reflects the component's location within the overall Caseflow hierarchy 
* [x] Write a separate story (within the same file) for each discrete variation of the component

### Testing Plan
- Login as `BVASYELLOW` or `BVAAABSHIRE`
- Go to `/queue` and pick any cases to look at the case details page

### CODE REVIEW

File: client/app/queue/utils.js [hunk 1]
```diff
@@ -306,7 +306,7 @@ export const prepareAppealForStore = (appeals) => {
       veteranFullName: appeal.attributes.veteran_full_name,
       veteranFileNumber: appeal.attributes.veteran_file_number,
       isPaperCase: appeal.attributes.paper_case,
-      sanitizedHearingRequestType: appeal.attributes.sanitized_hearing_request_type,
+      readableHearingRequestType: appeal.attributes.readable_hearing_request_type,
```
Comment (rubaiyat22) [inline client/app/queue/utils.js:309]: `sanitizedHearingRequestType` was not being used anywhere

**Review by tomas-nava — CHANGES_REQUESTED** [review]
This is looking great! I think, though, that if a hearing is scheduled, we want to display the type of that hearing, or the type of the most recently held hearing if it has a disposition. So if I've got a central or video hearing that's been converted to virtual, I should see "Virtual" as the hearing type.

File: app/models/appeal.rb [hunk 2]
```diff
@@ -505,6 +505,17 @@ def ready_for_bva_dispatch?
     false
   end
 
+  # if there hearings already scheduled, return the type of latest hearing
+  # if no hearings have been scheduled, determine type using cloesest_regional_office
+  # "Central" if closest_regional_office office is "C", "Video" otherwise
+  def readable_hearing_request_type
+    return hearings.last.readable_request_type if !hearings.empty?
```
[the remark this answers is not in the document]
Comment (rubaiyat22) [reply app/models/appeal.rb]: Tomas and I spoke async and decided that for now we should just evaluate request type based on the original request and not consider scheduled hearings. this [commit](https://github.com/department-of-veterans-affairs/caseflow/pull/15145/commits/b9d01e273cf874d2cb05ecf7252c969cc6bfd501) undoes some changes related to that. We will talk about this further with design and follow up from there.

File: spec/models/appeal_spec.rb [hunk 3]
```diff
@@ -1164,15 +1164,6 @@
...
-    context "there exists a scheduled hearing" do
-      let(:appeal) { create(:appeal) }
-      let!(:hearing) { create(:hearing, appeal: appeal) }
-
-      it "returns 'Central'" do
-        expect(appeal.readable_hearing_request_type).to eq("Central")
-      end
-    end
-
     context "no hearings have been scheduled" do
```
Comment (rubaiyat22) [inline spec/models/appeal_spec.rb:1167]: left this context as is for when we introduce evaluating request type based on scheduled hearings

**Review by tomas-nava — APPROVED** [review]
approve!

**Pull request merged**